io_scene_niftools/modules/nif_import/animation/material.py

```diff
@@ -59,62 +64,130 @@ def import_material_controllers(self, n_geom, b_material):
                                               ("specular_color", NifFormat.TargetColor.TC_SPECULAR)):
                 self.import_material_color_controller(b_material, n_material, b_channel, n_target_color)
 
-        self.import_material_uv_controller(b_material, n_geom)
+        self.import_uv_controller(b_material, n_geom)
+        self.import_tex_transform_controller(b_material, n_geom)
 
     def import_material_alpha_controller(self, b_material, n_material):
         # find alpha controller
-        n_alphactrl = math.find_controller(n_material, NifFormat.NiAlphaController)
-        if not n_alphactrl:
+        n_ctrl = math.find_controller(n_material, NifFormat.NiAlphaController)
+        if not n_ctrl:
             return
         NifLog.info("Importing alpha controller")
 
         b_mat_action = self.create_action(b_material, "MaterialAction")
-        fcurves = self.create_fcurves(b_mat_action, "niftools.emissive_alpha", range(3), n_alphactrl.flags)
-        interp = self.get_b_interp_from_n_interp(n_alphactrl.data.data.interpolation)
-        for key in n_alphactrl.data.data.keys:
-            self.add_key(fcurves, key.time, (key.value, key.value, key.value), interp)
+        n_ctrl_data = self.get_controller_data(n_ctrl)
+        interp = self.get_b_interp_from_n_interp(n_ctrl_data.interpolation)
+        times, keys = self.get_keys_values(n_ctrl_data.keys)
+        # key needs to be RGB due to current representation in blender
+        keys = [(v, v, v) for v in keys]
+        self.add_keys(b_mat_action, "niftools.emissive_alpha", range(3), n_ctrl.flags, times, keys, interp)
 
     def import_material_color_controller(self, b_material, n_material, b_channel, n_target_color):
         # find material color controller with matching target color
-        for ctrl in n_material.get_controllers():
-            if isinstance(ctrl, NifFormat.NiMaterialColorController):
-                if ctrl.get_target_color() == n_target_color:
-                    n_matcolor_ctrl = ctrl
+        for n_ctrl in n_material.get_controllers():
```

Comment:
It would be nice to move these to processable entities like we do for props in the future

Reply:
An there's our problem. I'm not sure if the processor approach works for materials. Maybe once everything is standardized. But so far the 'unpredictable' order of processing causes trouble for the node system.

Maybe we should change the code to first collect nodes, then build node tree, finally animate the nodes.

Reply:
Yeah we had a resolve a similiar issue when processing Ni**Properties to cater for when a material existed or not.
At least it will get you thinking 🤔

Reply:
Exactly.
We'll basically need several phases:
1) go over all nodes, collect properties, see which n_geoms have unique combinations of properties
2) for each n_geom, create b_mat and b_nodes from props as needed
3) for each n_geom, create b_animations from material controllers as needed
4) link the nodes for each b_mat

Reply:
Probably need to generate hashCodes at each level, feeds into a few of the other conversations we have been having in the PR. 

Solution for now 👍 
![image](https://user-images.githubusercontent.com/525919/170677472-020507af-d7fc-4a4c-8207-90b984c36305.png)